public/app/features/library-panels/components/AddLibraryPanelModal/AddLibraryPanelModal.tsx

```diff
@@ -12,11 +15,34 @@ interface AddLibraryPanelContentsProps {
 export const AddLibraryPanelContents = ({ panel, initialFolderId, onDismiss }: AddLibraryPanelContentsProps) => {
   const [folderId, setFolderId] = useState(initialFolderId);
   const [panelTitle, setPanelTitle] = useState(panel.title);
+  const [debouncedPanelTitle, setDebouncedPanelTitle] = useState(panel.title);
+  const [waiting, setWaiting] = useState(false);
+
+  useEffect(() => setWaiting(true), [panelTitle]);
+  useDebounce(() => setDebouncedPanelTitle(panelTitle), 350, [panelTitle]);
+  const isValidTitle = useAsync(async () => {
+    try {
+      const libraryPanels = (await getLibraryPanelByName(panelTitle)).filter((lp) => lp.folderId === folderId);
+      return libraryPanels.length === 0;
+    } catch (err) {
+      err.isHandled = true;
+      return true;
+    } finally {
+      setWaiting(false);
+    }
+  }, [debouncedPanelTitle]);
+
+  const invalidInput =
+    !isValidTitle?.value && isValidTitle.value !== undefined && panelTitle === debouncedPanelTitle && !waiting;
```

Comment:
Settled on the use of `waiting` here to avoid a small bit of jank that I couldn't find a way to work around otherwise.
Initially I had wanted to use `!isValidTitle.loading` as one of the conditions here, but this resulted in a flash between valid/invalid styles when editing an invalid title to a valid one.
The reason this happens is that, when the user starts typing `panelTitle` no longer equals `debouncedPanelTitle`, so `invalidInput` becomes false, and the valid styling is used.
When the user stops typing, after 350ms `debouncedPanelTitle` equals `panelTitle`. I had hoped then, that the `useAsync` callback would execute and set `isValidTitle.loading` to true, thus keeping `invalidInput` false, but I suspect because of some event loop weirdness this doesn't happen for a frame or two, and so we get a flash of invalid styling before `isValidTitle.loading` settles on being true.
Anyway, if anyone knows a nicer way to do this, let me know.